sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/client.py

```diff
@@ -105,15 +130,21 @@ def encrypt(self, algorithm, plaintext, **kwargs):
 
         """
 
-        result = self._client.encrypt(
-            vault_base_url=self._key_id.vault_url,
-            key_name=self._key_id.name,
-            key_version=self._key_id.version,
-            algorithm=algorithm,
-            value=plaintext,
-            **kwargs
-        )
-        return EncryptResult(key_id=self.key_id, algorithm=algorithm, ciphertext=result.result, authentication_tag=None)
+        local_key = self._get_local_key()
+        if local_key:
+            if "encrypt" not in self._allowed_ops:
```

Comment:
Can't `self_.allowed_ops` currently be None in some cases?

Reply:
Yes but in that case `local_key is None`. Your feedback tells me this is too complex, so I'll assign `_allowed_ops` an empty `frozenset`.

Reply:
+1 I wondered if that might be the case, but verifying the same is a lot harder to do